lib/scope.go

```diff
@@ -20,9 +22,46 @@ import (
 type scope struct {
 	ctx  context.Context
 	inst *Instance
+	pro  *profile.Profile
 	// TODO(dustmop): Additional information, such as user identity, their profile, keys
 }
 
+func newScope(ctx context.Context, inst *Instance) (scope, error) {
+	var pro *profile.Profile
+	// TODO(b5): shouldn't need this nil checking
+	if inst != nil && inst.profiles != nil {
```

Comment:
Could we move this part into its own helper function? Would suggest `pro := activeProfile(inst)`. It'll be nice to keep the `newScope` constructor itself short and clear, as I can only imagine it's going to be adding much more functionality soon.